pkg/api/validation/events.go

```diff
@@ -25,11 +25,20 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	// TODO: event.InvolvedObject is not versioned. References need to be normalizd to internal when converted from external and vice versa.
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
Context is here: https://github.com/GoogleCloudPlatform/kubernetes/pull/3613 and here https://github.com/GoogleCloudPlatform/kubernetes/pull/3944

----- Original Message -----
> >  	if !util.IsDNSSubdomain(event.Namespace) {
> >  		allErrs = append(allErrs, errs.NewFieldInvalid("namespace",
> >  		event.Namespace, ""))
> >  	}
> > +	// TODO: event.InvolvedObject is not versioned. References need to be
> > normalizd to internal when converted from external and vice versa.
> > +	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind ==
> > "Node" {
> 
> Yes, I think you're right that under the current system, we can't validate
> this here.
> 
> In fact this kinda breaks events-- do we allow events to be both namespaced
> AND root objects?
> 
> However, I'm questioning the decision to allow non-namespaced resources. It
> greatly complicates logic like this and in #4304, which I'm working on as my
> get-back-up-to-speed issue. Why did we not just declare all such objects to
> be in the "default" namespace? Visibility concerns?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4396/files#r24930925